lib/filterx/filterx-metrics-labels.c

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright (c) 2024 Axoflow
+ * Copyright (c) 2024 Attila Szakacs <attila.szakacs@axoflow.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "filterx-metrics-labels.h"
+#include "filterx-eval.h"
+#include "expr-literal.h"
+#include "expr-literal-generator.h"
+#include "object-string.h"
+#include "object-dict-interface.h"
+#include "metrics/metrics-tls-cache.h"
+#include "stats/stats-cluster-single.h"
+#include "scratch-buffers.h"
+
+#include <string.h>
+
+typedef struct _FilterXMetricsLabel
+{
+  gchar *name;
+  struct
+  {
+    FilterXExpr *expr;
+    gchar *str;
+  } value;
```

Comment:
isn't this a union? my understanding that only one of these is set at a time.

Reply:
scratch that, the only thing that indicates if one is set if str is NULL or not.